eva-test/src/main/kotlin/com/razz/eva/test/db/DatabaseContainer.kt

```diff
@@ -0,0 +1,98 @@
+package com.razz.eva.test.db
+
+import com.razz.eva.test.db.DockerImageName.PostgrePartmanImage16
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import org.testcontainers.containers.PostgreSQLContainer
+import org.testcontainers.containers.PostgreSQLContainer.POSTGRESQL_PORT
+import java.lang.Runtime.getRuntime
+import kotlin.math.ceil
+import org.testcontainers.utility.DockerImageName as TestcontainersDockerImageName
+
+data class DatabaseContainer(
+    val pgContainer: PostgreDockerContainer,
+) {
+
+    private val localPools = mutableMapOf<String, HikariDataSource>()
+
+    val managementPool = HikariConfig().run {
+        jdbcUrl = pgContainer.jdbcUrl
+        username = "test"
+        password = "test"
+        maximumPoolSize = 2
+        HikariDataSource(this)
+    }
+
+    fun localPool(dbName: String, size: Int) = synchronized(pgContainer) {
+        localPools.getOrPut(dbName) {
+            HikariConfig().run {
+                jdbcUrl = jdbcUrl(dbName)
+                username = "test"
+                password = "test"
+                maximumPoolSize = size
+                initializationFailTimeout = -1
+                HikariDataSource(this)
```

Comment:
all args should be named yo

Reply:
We return this source from run method yo

Reply:
yo